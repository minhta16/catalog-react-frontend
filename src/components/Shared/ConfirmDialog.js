import React from 'react';
import PropTypes from 'prop-types';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  Typography,
  Button,
  DialogActions,
} from '@material-ui/core';

/**
 * This component creates a confirm dialog which calls onConfirm when the user clicks confirm and onCancel when the user clicks cancel
 */
export function ConfirmDialog(props) {
  const { title, contentText, open, onCancel, onClose, onConfirm } = props;
  return (
    <Dialog open={open} onClose={onClose}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <Typography variant="body1">{contentText}</Typography>
      </DialogContent>
      <DialogActions>
        <Button id="confirm-dialog-confirm" onClick={onConfirm} color="primary">
          Confirm
        </Button>
        <Button id="confirm-dialog-cancel" onClick={onCancel}>
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
}

ConfirmDialog.propTypes = {
  title: PropTypes.string,
  contentText: PropTypes.string,
  open: PropTypes.bool,
  onCancel: PropTypes.func,
  onConfirm: PropTypes.func,
  onClose: PropTypes.func,
};

ConfirmDialog.defaultProps = {
  title: '',
  contentText: '',
  open: false,
  onCancel: () => {},
  onConfirm: () => {},
  onClose: () => {},
};

export default ConfirmDialog;
