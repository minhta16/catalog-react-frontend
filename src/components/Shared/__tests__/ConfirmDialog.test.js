/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { Dialog } from '@material-ui/core';
import { ConfirmDialog } from '../ConfirmDialog';

describe('components/LoginDialog', () => {
  let wrapper;
  let props;
  // let warning;

  const setup = () => {
    wrapper = shallow(<ConfirmDialog {...props} />);
  };

  beforeEach(() => {
    props = {
      title: 'title',
      contentText: 'text',
      open: true,
    };
  });

  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onConfirm on button click', () => {
    props.onConfirm = jest.fn();
    setup();
    const confirm = wrapper.find('#confirm-dialog-confirm');
    confirm.simulate('click');
    expect(props.onConfirm).toHaveBeenCalled();
  });

  it('should call onCancel on button click', () => {
    props.onCancel = jest.fn();
    setup();
    const cancel = wrapper.find('#confirm-dialog-cancel');
    cancel.simulate('click');
    expect(props.onCancel).toHaveBeenCalled();
  });

  it('should call onClose on button click', () => {
    props.onClose = jest.fn();
    setup();
    const dialog = wrapper.find(Dialog);
    dialog.simulate('close');
    expect(props.onClose).toHaveBeenCalled();
  });

  it('should define default props', () => {
    setup();
    expect(ConfirmDialog.defaultProps.onConfirm).toBeDefined();
    expect(ConfirmDialog.defaultProps.onClose).toBeDefined();
    expect(ConfirmDialog.defaultProps.onCancel).toBeDefined();
  });

  it('should return undefined for default functions', () => {
    setup();
    expect(ConfirmDialog.defaultProps.onConfirm()).toBe(undefined);
    expect(ConfirmDialog.defaultProps.onClose()).toBe(undefined);
    expect(ConfirmDialog.defaultProps.onCancel()).toBe(undefined);
  });
});
