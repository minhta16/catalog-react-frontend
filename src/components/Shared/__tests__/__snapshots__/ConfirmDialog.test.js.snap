// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoginDialog should render correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))>
    title
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      text
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      id="confirm-dialog-confirm"
      onClick={[Function]}
    >
      Confirm
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      id="confirm-dialog-cancel"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
